internal/form/submit/service.go

```diff
@@ -79,14 +104,21 @@ func (s *Service) Submit(ctx context.Context, formID uuid.UUID, userID uuid.UUID
 		}
 
 		if !found {
-			err := fmt.Errorf("question with ID %s not found in form %s", ans.QuestionID, formID)
-			validationErrors = append(validationErrors, err)
+			validationErrors = append(validationErrors, fmt.Errorf("question with ID %s not found in form %s", ans.QuestionID, formID))
+		}
+	}
+
+	// Check for required questions that were not answered
+	for _, q := range questions {
+		if q.Question().Required && !answeredQuestionIDs[q.Question().ID.String()] {
+			validationErrors = append(validationErrors, fmt.Errorf("question ID %s is required but not answered", q.Question().ID.String()))
```

Comment:
Please update the method documentation to include the new behavior that validates whether required answers are filled.